runtime/riscv.S

```diff
@@ -16,12 +16,12 @@
 /* Asm part of the runtime system, RISC-V processor, 64-bit mode */
 /* Must be preprocessed by cpp */
 
-#define ARG_DOMAIN_STATE_PTR t1
+#define ARG_DOMAIN_STATE_PTR t0
 #define DOMAIN_STATE_PTR s0
 #define TRAP_PTR s1
 #define ALLOC_PTR s10
 #define ALLOC_LIMIT s11
-#define TMP t0
+#define TMP t1
 #define ARG t2
```

Comment:
I'm lost in the way you juggle t0, t1, t2 here, but I hope this was tested and passed.

Reply:
Yes, it was tested, and passed :)

Briefly, both `t0` and `t1` were reserved as temporaries before the change. In the assembly stubs, `TMP` is used as a general "scratch" register and `ARG_DOMAIN_STATE_PTR` is only used in a couple of specific places to pass the domain state pointer among the `caml_callback*_asm` routines and the code of `caml_start_program`. After the change, `t0` can be used by the register allocator and `t1` remains reserved. So I changed `TMP` to use `t1` (which remains generally reserved), and `ARG_DOMAIN_STATE_PTR` (which is only used in a couple of places) to `t0`.

It is possible this change was unnecessary, but it was easier for me to check the few uses of `ARG_DOMAIN_STATE_PTR` rather than the many uses of `TMP`.